internal/transport/http2_client.go

```diff
@@ -1292,6 +1293,19 @@ func (t *http2Client) reader() {
 	}
 }
 
+// shouldKeepaliveBeActive returns true if the keepalive goroutine should be
+// running, and false otherwise. The decision is based on the number of active
+// streams and the `PermitWithoutStream` configuration parameter.
+// t.mu should be held when this method is called.
+func (t *http2Client) shouldKeepaliveBeActive() bool {
+	if t.state == closing {
```

Comment:
I understand the need for this check. But it shouldn't be done in this function. Because semantically, if transport's state is closing, then keepalive shouldn't be active, but should be returning as well.

Reply:
Moved the check to the keepalive goroutine, and now this function seems unnecessary. So, removed the function as well, and reverted to checking the condition for dormancy as it used to be before this PR.